src/core/weightedGraph.js

```diff
@@ -76,3 +77,108 @@ export function overrideWeights(
   });
   return {graph: wg.graph, weights};
 }
+
+export type NodeWithWeight = {|
+  +node: Node,
+  +weight: ?Weights.NodeWeight,
+|};
+
+export type EdgeWithWeight = {|
+  +edge: Edge,
+  +weight: ?Weights.EdgeWeight,
+|};
+
+export type WeightedGraphComparison = {|
+  +weightedGraphsAreEqual: boolean,
+  +uniqueNodesInFirst: $ReadOnlyArray<NodeWithWeight>,
+  +uniqueNodesInSecond: $ReadOnlyArray<NodeWithWeight>,
+  +nodeTuplesWithDifferences: $ReadOnlyArray<[NodeWithWeight, NodeWithWeight]>,
+  +uniqueEdgesInFirst: $ReadOnlyArray<EdgeWithWeight>,
+  +uniqueEdgesInSecond: $ReadOnlyArray<EdgeWithWeight>,
+  +edgeTuplesWithDifferences: $ReadOnlyArray<[EdgeWithWeight, EdgeWithWeight]>,
+|};
```

Comment:
This comparison is incapable of representing the case where the WeightedGraphs differ in prefix-set weights (i.e. weights that do not correspond to a specific node in the graph)